frontend/packages/console-plugin-sdk/src/typings/dashboards.ts

```diff
@@ -31,6 +33,25 @@ namespace ExtensionProperties {
     /** The Prometheus query */
     query: string;
   }
+
+  export interface DashboardsTab {
+    /** The tab's ID which will be used as href within dashboards page */
+    id: string;
+
+    /** The tab title */
+    title: string;
+  }
+
+  export interface DashboardsCard {
+    /** The tab's ID where this card should be rendered */
+    tabID: string;
+
+    /** The card position in the tab */
+    position: GridPosition;
+
+    /** The card component to render */
+    component: React.ComponentType<any>;
```

Comment:
cc @spadgett @christianvogt 

Directly referencing a React component from within a plugin's entry module will make it part of the Console initial chunk, which is already too big.

The default strategy for referencing plugin-contributed React components should be

```ts
Promise<TypeOfComponent>
```

and using either

```ts
() => import('./path/to/module' /* webpackChunkName: "chunk-name" */).then(m => m.MyComponent)
```

or

```ts
async () => (await import('./path/to/module' /* webpackChunkName: "chunk-name" */)).MyComponent
```

to provide those components.

Please add the following into `typings/index.ts`:

```ts
type LazyLoader<T> = () => Promise<React.ComponentType<T>>;
```

and use it in both `typings/pages.ts` and `typings/dashboards.ts`.

Reply:
I cant move `LazyLoader` type to `typings/index.ts` because that results in a circular dependency. Moved it to `typings/types.ts`.